ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -1256,6 +1254,24 @@ check_parent_attrs_against_processing(enum pkcs11_mechanism_id proc_id,
 
 		return PKCS11_CKR_KEY_FUNCTION_NOT_PERMITTED;
 
+	case PKCS11_CKM_AES_ECB_ENCRYPT_DATA:
+	case PKCS11_CKM_AES_CBC_ENCRYPT_DATA:
+		if (key_class != PKCS11_CKO_SECRET_KEY &&
+		    key_type != PKCS11_CKK_AES)
+			return PKCS11_CKR_KEY_FUNCTION_NOT_PERMITTED;
```

Comment:
this part belongs to the 2nd commit in the series.

Reply:
The key check is already done for other AES mechanisms. So in the 2nd commit, PKCS11_CKM_AES_CBC_ENCRYPT_DATA was added along with other mechanisms in the switch for this check. Here as we are adding an additional check for encrypt, I separated it out in this commit.

Reply:
oh, sorry, I completely messed my mind :confused:
Yes, sorry, as is it just ok.